test/local/db_tests.js

```diff
@@ -206,6 +207,120 @@ test(
 )
 
 test(
+  'device registration',
+  function (t) {
+    var sessionTokenId
+    var deviceInfo = {
+      id: crypto.randomBytes(16),
+      name: '',
+      type: 'mobile',
+      pushCallback: 'https://foo/bar',
+      pushPublicKey: crypto.randomBytes(32)
+    }
+    return dbConn.then(function (db) {
+      return db.emailRecord(ACCOUNT.email)
+        .then(function (emailRecord) {
+          return db.createSessionToken(emailRecord, 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:44.0) Gecko/20100101 Firefox/44.0')
+        })
+        .then(function (sessionToken) {
+          sessionTokenId = sessionToken.tokenId
+          return db.updateDevice(ACCOUNT.uid, sessionTokenId, deviceInfo)
+            .then(function () {
+              t.fail('updating a non-existent device should have failed')
+            }, function (err) {
+              t.pass('updating a non-existent device failed')
+              t.equal(err.errno, 123, 'err.errno === 123')
+            })
+        })
+        .then(function () {
+          return db.deleteDevice(ACCOUNT.uid, deviceInfo.id)
+            .then(function () {
+              t.fail('deleting a non-existent device should have failed')
+            }, function (err) {
+              t.pass('deleting a non-existent device failed')
+              t.equal(err.errno, 123, 'err.errno === 123')
+            })
+        })
+        .then(function () {
+          return db.devices(ACCOUNT.uid)
+            .catch(function () {
+              t.fail('getting devices should not have failed')
+            })
```

Comment:
Pretty much, yep. During development, I found it really frustrating when an error would cause the promise chain to fail miles away from its point of origin. Adding these local catches in made it much quicker debug what I'd broken. Because they were useful to me then, I figured they were worth keeping around. Happy to take them out if you don't like them though.